controllers/nodemaintenance_controller_test.go

```diff
@@ -2,338 +2,346 @@ package controllers
-			Expect(err).ToNot(HaveOccurred())
-		}
-
-		// Mock request to simulate Reconcile() being called on an event for a watched resource .
-		req = reconcile.Request{
-			NamespacedName: types.NamespacedName{
-				Name: nm.ObjectMeta.Name,
-			},
+		if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(testNs), &corev1.Namespace{}); err != nil {
+			Expect(k8sClient.Create(ctx, testNs)).To(Succeed())
 		}
+		nodeOne = getNode(taintedNodeName)
 	})
-
-	AfterEach(func() {
-		stopTestEnv()
-	})
-
-	Context("Initialization test", func() {
-
-		It("Node maintenance should be initialized properly", func() {
-			_ = r.initMaintenanceStatus(nm)
-			maintenance := &nodemaintenanceapi.NodeMaintenance{}
-			err := k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nm), maintenance)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(maintenance.Status.Phase).To(Equal(nodemaintenanceapi.MaintenanceRunning))
-			Expect(len(maintenance.Status.PendingPods)).To(Equal(2))
-			Expect(maintenance.Status.EvictionPods).To(Equal(2))
-			Expect(maintenance.Status.TotalPods).To(Equal(2))
-			Expect(maintenance.Status.DrainProgress).To(Equal(0))
-			Expect(maintenance.Status.LastUpdate.IsZero()).To(BeFalse())
-		})
-
-		It("owner ref should be set properly", func() {
-			_ = r.initMaintenanceStatus(nm)
-			maintenance := &nodemaintenanceapi.NodeMaintenance{}
-			err := k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nm), maintenance)
-			node := &corev1.Node{}
-			err = k8sClient.Get(context.TODO(), client.ObjectKey{Name: "node01"}, node)
-			Expect(err).ToNot(HaveOccurred())
-			r.setOwnerRefToNode(maintenance, node)
-			Expect(len(maintenance.ObjectMeta.GetOwnerReferences())).To(Equal(1))
-			ref := maintenance.ObjectMeta.GetOwnerReferences()[0]
-			Expect(ref.Name).To(Equal(node.ObjectMeta.Name))
-			Expect(ref.UID).To(Equal(node.ObjectMeta.UID))
-			Expect(ref.APIVersion).To(Equal(node.TypeMeta.APIVersion))
-			Expect(ref.Kind).To(Equal(node.TypeMeta.Kind))
-			r.setOwnerRefToNode(maintenance, node)
-			Expect(len(maintenance.ObjectMeta.GetOwnerReferences())).To(Equal(1))
+	Context("Functionality test", func() {
+		Context("Testing initMaintenanceStatus", func() {
+			var nm *nodemaintenanceapi.NodeMaintenance
+			BeforeEach(func() {
+				Expect(k8sClient.Create(ctx, nodeOne)).To(Succeed())
+				DeferCleanup(k8sClient.Delete, ctx, nodeOne)
+
+				podOne, podTwo = getTestPod("test-pod-1", taintedNodeName), getTestPod("test-pod-2", taintedNodeName)
+				Expect(k8sClient.Create(ctx, podOne)).To(Succeed())
+				Expect(k8sClient.Create(ctx, podTwo)).To(Succeed())
+				DeferCleanup(cleanupPod, ctx, podOne)
+				DeferCleanup(cleanupPod, ctx, podTwo)
+				nm = getTestNM("node-maintenance-cr-initMaintenanceStatus", taintedNodeName)
+			})
+			When("Status was initalized", func() {
+				It("should be set for running with 2 pods to drain", func() {
+					Expect(initMaintenanceStatus(nm, r.drainer, r.Client)).To(HaveOccurred())
```

Comment:
Expecting an error here is a nice indication IMHO that the implementation has an issue. Here: it would be better to separate setting status fields, and doing the actual update. Out of scope for this PR though.